opennlp-tools/src/main/java/opennlp/tools/ml/naivebayes/Probabilities.java

```diff
@@ -146,14 +147,15 @@ public Double getLog(T t) {
       return normalised;
     Map<T, Double> temp = createMapDataStructure();
     double sum = 0;
-    for (T t : map.keySet()) {
-      Double p = map.get(t);
+    for (Entry<T, Double> entry : map.entrySet()) {
+      final Double p = entry.getValue();
       if (p != null) {
         sum += p;
       }
     }
-    for (T t : temp.keySet()) {
-      Double p = temp.get(t);
+    for (Entry<T, Double> entry : temp.entrySet()) {
+      final T t = entry.getKey();
```

Comment:
Don't make it final. No need to declare variable t, maybe looks nicer with Java Streams